src/libraries/System.IO.FileSystem.Watcher/src/System/IO/FileSystemEventArgs.cs

```diff
@@ -19,7 +19,7 @@ public FileSystemEventArgs(WatcherChangeTypes changeType, string directory, stri
         {
             _changeType = changeType;
             _name = name;
-            _fullPath = Combine(directory, name);
+            _fullPath = Path.GetFullPath(Combine(directory, name));
```

Comment:
`Combine("", "")` will return `\`, which I think is a bug since a single separator refers to the root folder.
We should start using `Path.Join` instead to avoid the code duplication.

```suggestion
            _fullPath = Path.GetFullPath(Path.Join(directory, name));
```

Reply:
Done.